internal/clients/http/azure/azure_service_client.go

```diff
@@ -0,0 +1,147 @@
+	}
+
+	return &serviceClient{
+		credential: identityClient,
+	}, nil
+}
+
+func (c *serviceClient) RegisterInstanceTypes(ctx context.Context, instanceTypes *clients.RegisteredInstanceTypes, regionalTypes *clients.RegionalTypeAvailability) error {
+	restricted := make(map[armcompute.ResourceSKURestrictionsReasonCode]int, 0)
+
+	skuClient, err := armcompute.NewResourceSKUsClient(config.Azure.SubscriptionID, c.credential, nil)
+	if err != nil {
+		return fmt.Errorf("unable to generate types: %w", err)
+	}
+	skuPager := skuClient.NewListPager(nil)
+	for skuPager.More() {
+		nextResult, pageErr := skuPager.NextPage(ctx)
+		if pageErr != nil {
+			return fmt.Errorf("unable to generate types: %w", pageErr)
+		}
+		for _, v := range nextResult.Value {
+			if *v.ResourceType != "virtualMachines" {
+				continue
+			}
+			instanceType := clients.InstanceType{
+				Name:        clients.InstanceTypeName(*v.Name),
+				AzureDetail: &clients.InstanceTypeDetailAzure{},
+			}
+			vcpusPerCore := int32(0)
+			for _, c := range v.Capabilities {
+				if *c.Name == "CpuArchitectureType" {
+					instanceType.Architecture, pageErr = clients.MapArchitectures(ctx, *c.Value)
+					if pageErr != nil {
+						return fmt.Errorf("unable to generate types: %w", pageErr)
+					}
+				}
+				if *c.Name == "vCPUs" {
+					vcpus, vcpuErr := strconv.ParseInt(*c.Value, 10, 32)
+					if vcpuErr != nil {
+						return fmt.Errorf("unable to generate types: %w", vcpuErr)
+					}
+					instanceType.VCPUs = int32(vcpus)
+				}
+				if *c.Name == "vCPUsPerCore" {
+					value, vcpupcErr := strconv.ParseInt(*c.Value, 10, 32)
+					if vcpupcErr != nil {
+						return fmt.Errorf("unable to generate types: %w", vcpupcErr)
+					}
+					vcpusPerCore = int32(value)
+				}
+				if *c.Name == "MaxResourceVolumeMB" {
+					mbs, volErr := strconv.Atoi(*c.Value)
+					if volErr != nil {
+						return fmt.Errorf("unable to generate types: %w", volErr)
+					}
+					instanceType.SetEphemeralStorageFromMB(int64(mbs))
+				}
+				// Appears to be GiB not GB.
+				if *c.Name == "MemoryGB" {
+					memoryGB, memErr := strconv.ParseFloat(*c.Value, 64)
+					if memErr != nil {
+						return fmt.Errorf("unable to generate types: %w", memErr)
+					}
+					instanceType.MemoryMiB = int64(memoryGB * 1000)
+				}
+				if *c.Name == "HyperVGenerations" {
+					if strings.Contains(*c.Value, "V1") {
+						instanceType.AzureDetail.GenV1 = true
+					}
+					if strings.Contains(*c.Value, "V2") {
+						instanceType.AzureDetail.GenV2 = true
+					}
+				}
+			}
+			if vcpusPerCore != 0 {
+				instanceType.Cores = instanceType.VCPUs / vcpusPerCore
+			} else {
+				// Some types have no HT (HPC AMD CPUs) thus vcpus per core is zero:
+				// Standard_B16ms, Standard_B20ms, Standard_HB60rs, Standard_HC44rs,
+				// Standard_M208ms_v2, Standard_M208s_v2, Standard_M416ms_v2,
+				// Standard_M416s_v2 and Standard_PB6s
+				instanceType.Cores = instanceType.VCPUs
+			}
```

Comment:
```suggestion
			instanceType.Cores = instanceType.VCPUs
			if vcpusPerCore != 0 {
				instanceType.Cores /= vcpusPerCore
			}
```

This is also possible, but if we need the comments, a helper function might also come in handy.

Reply:
Sure, I shortened the comment.